File: src/eQuantic.UI.Runtime/src/shared/components/Banner.ts
import { Box, BoxStyle, BuildContext, Column, CornerRadii, EdgeInsets, Flexible, Icon, Pressable, Row, SizeValue, StatelessComponent, Text, TypeStyle } from "../runtime-exports";

export class Banner extends StatelessComponent {
    declare status: string;
    declare title: string;
    declare body: any;
    declare primaryAction: any;
    declare secondaryAction: any;
    declare onDismiss: (() => void) | null;
    constructor(status?: any, title?: any, body: any = null, props?: any) {
        super();
        if (status !== undefined) this.status = status;
        if (title !== undefined) this.title = title;
        if (body !== undefined) this.body = body;
        if (this.status === undefined) this.status = 'primary';
        this.status = status;this.title = title;this.body = body;
        if (props && typeof props === 'object') Object.assign(this, props);
    }

    build(context: BuildContext) {
        let tint = context.theme.colors(this.status);let glyph = (() => { const _s = this.status; if (_s === 'success') return 'checkCircle'; if (_s === 'warning') return 'warning'; if (_s === 'destructive') return 'error'; return 'info'; })();let column = new Column(4);column.add(new Text(this.title, 'caption', tint.onSubtle, 2, 'start', false, false, null, { styleOverride: new TypeStyle(13, 18, 'semiBold', 0, 1.3) }));if (this.body != null) {column.add(new Text(this.body, 'caption', tint.onSubtle, 4, 'start', false, false, null, { styleOverride: new TypeStyle(13, 18, 'regular', 0, 1.3) }));}if (this.primaryAction != null || this.secondaryAction != null) {let actions = new Row(8);if (this.primaryAction != null) actions.add(this.primaryAction);if (this.secondaryAction != null) actions.add(this.secondaryAction);column.add(actions);}let content = new Row(10, 'start', 'center', false, null, null, { cross: 'start' });content.add(new Icon(glyph, 20, tint.onSubtle));content.add(new Flexible(column));if (this.onDismiss != null) {content.add(new Pressable(new Icon('close', 20, tint.onSubtle), this.onDismiss, { label: 'Dismiss' }));}return new Box(new BoxStyle({ width: SizeValue.fill, padding: new EdgeInsets(14, 12, 14, 12), background: tint.subtle, cornerRadius: new CornerRadii(context.theme.shape('large')) }), content);
    }

}

File: src/eQuantic.UI.Components/Banner.cs
using eQuantic.UI.Primitives;

namespace eQuantic.UI.Components;

/// <summary>
/// The design system's Banner / Alert (spec B18): an inline status surface that scrolls with the
/// content it describes — Radius.Lg, the status variant's Subtle pair fill, 12/14 padding, 13dp text
/// with a bold lead-in instead of a separate title row, ≤ 2 text actions. v1 fence: the bold lead-in renders
/// as two Text nodes until rich-text spans exist.
/// </summary>
public sealed class Banner : StatelessComponent
{
    public Banner(Variant status, string title, string? body = null)
    {
        Status = status;
        Title = title;
        Body = body;
    }

    public Variant Status { get; init; }
    public string Title { get; init; }
    public string? Body { get; init; }

    /// <summary>Up to two text actions (spec) — rendered as Link buttons in the status tint.</summary>
    public Button? PrimaryAction { get; init; }
    public Button? SecondaryAction { get; init; }

    /// <summary>Dismiss affordance (spec: 20dp close, 48dp hit) — fires when the user closes the banner;
    /// the OWNER removes it from the tree (banners hold no internal state).</summary>
    public Action? OnDismiss { get; init; }

    public override VisualNode Build(ComponentContext context)
    {
        var tint = context.Theme.Colors(Status);
        var glyph = Status switch
        {
            Variant.Success => Icons.CheckCircle,
            Variant.Warning => Icons.Warning,
            Variant.Destructive => Icons.Error,
            _ => Icons.Info,
        };

        var column = new Column(gap: Space.S1);
        column.Add(new Text(Title, TypeRole.Caption, tint.OnSubtle, maxLines: 2)
        {
            StyleOverride = new TypeStyle(13, 18, FontWeight.SemiBold, 0, 1.3f),
        });
        if (Body != null)
        {
            column.Add(new Text(Body, TypeRole.Caption, tint.OnSubtle, maxLines: 4)
            {
                StyleOverride = new TypeStyle(13, 18, FontWeight.Regular, 0, 1.3f),
            });
        }
        if (PrimaryAction != null || SecondaryAction != null)
        {
            var actions = new Row(gap: Space.S2);
            if (PrimaryAction != null) actions.Add(PrimaryAction);
            if (SecondaryAction != null) actions.Add(SecondaryAction);
            column.Add(actions);
        }

        // Spec B18: status icon Md 20 + gap 10, top-aligned with the lead-in.
        var content = new Row(gap: 10) { Cross = CrossAlign.Start };
        content.Add(new Icon(glyph, IconSize.Dense, tint.OnSubtle));
        content.Add(new Flexible(column));
        if (OnDismiss != null)
        {
            content.Add(new Pressable(new Icon(Icons.Close, IconSize.Dense, tint.OnSubtle), OnDismiss)
            {
                Label = SdkStrings.Dismiss,
            });
        }

        return new Box(new BoxStyle
        {
            Width = SizeValue.Fill,
            Padding = new EdgeInsets(14, 12, 14, 12),
            Background = tint.Subtle,
            CornerRadius = new CornerRadii(context.Theme.Shape(ShapeScale.Large)),
        }, content);
    }
}
